src/utils/dep-warn.js

```diff
@@ -0,0 +1,33 @@
+// 'new' is optional
+const DEPRECATED_PROPS = [
+  {old: 'onChangeViewport', new: 'onViewportChange'},
+  {old: 'perspectiveEnabled', new: 'dragRotate'},
+  {old: 'onHoverFeatures', new: 'onHover'},
+  {old: 'onClickFeatures', new: 'onClick'}
+];
+
+function getDeprecatedText(name) {
+  return `react-map-gl: \`${name}\` is deprecated and will be removed in a later version.`;
+}
+
+function getNewText(name) {
+  return `Use \`${name}\` instead.`;
+}
+
+/**
+ * Checks props object for any prop that is deprecated and insert a console
+ * warning to the user. This will also print out the recommended new prop/API
+ * if one exists.
+ */
+export default function checkDeprecatedProps(props = {}) {
+  /* eslint-disable no-console, no-undef */
+  DEPRECATED_PROPS.forEach((depProp) => {
+    if (props.hasOwnProperty(depProp.old)) {
```

Comment:
It's the best way to check if an object has a property that it owns (vs. inherited). We'd only want to log a warning if the user is trying to use an old prop imo.